lib/rubocop/cop/rspec/expect_change.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for consistent style of change matcher.
+      #
+      # Enforces either `and_return` or block-style return in the cases
+      # where the returned value is constant. Ignores dynamic returned values
+      # are the result would be different
+      #
+      # This cop can be configured using the `EnforcedStyle` option
+      #
+      # @example `EnforcedStyle: block`
+      #   # bad
+      #   expect(run).to change(Foo, :bar)
+      #
+      #   # good
+      #   expect(run).to change { Foo.bar }
+      #
+      # @example `EnforcedStyle: method_call`
+      #   # bad
+      #   expect(run).to change { Foo.bar }
+      #   expect(run).to change { foo.baz }
+      #
+      #   # good
+      #   expect(run).to change(Foo, :bar)
+      #   expect(run).to change(foo, :baz)
+      #   # also good when there are arguments or chained method calls
+      #   expect(run).to change { Foo.bar(:count) }
+      #   expect(run).to change { user.reload.name }
+      #
+      class ExpectChange < Cop
+        include ConfigurableEnforcedStyle
+
+        MSG_BLOCK = 'Prefer `change(%<obj>s, :%<attr>s)`.'.freeze
+        MSG_CALL = 'Prefer `change { %<obj>s.%<attr>s }`.'.freeze
+
+        def_node_matcher :expect_change_with_block, <<-PATTERN
+          (block
+            (send nil? :change)
+            (args)
+            (send ({const send} nil? $_) $_)
+          )
+        PATTERN
+
+        def_node_matcher :expect_change_with_arguments, <<-PATTERN
+          (send nil? :change ({const send} nil? $_) (sym $_))
+        PATTERN
+
+        def on_send(node)
+          return unless style == :block
+
+          expect_change_with_arguments(node) do |receiver, message|
+            add_offense(
+              node,
+              message: format(MSG_CALL, obj: receiver, attr: message)
+            )
+          end
+        end
+
+        def on_block(node)
+          return unless style == :method_call
```

Comment:
Add a blank line after this one (or remove blank line after `return` in `#on_send`). I wonder if it would be possible to have a cop check where blank lines should be.

Reply:
That would be a nice cop to add to RuboCop - I prefer having a blank line after early return